src/_cffi_src/openssl/callbacks.py

```diff
@@ -118,4 +95,31 @@
     }
     return 1;
 }
+
+typedef struct {
+    char *password;
+    int length;
+    int called;
+    int error;
+    int maxsize;
+} CRYPTOGRAPHY_ST_PW_CB;
+
+int _cryptography_pem_password_cb(char *buf, int size,
+                                  int rwflag, void *userdata) {
+    /* The password cb is only invoked if OpenSSL decides the private
+       key is encrypted. So this path only occurs if it needs a password */
+    CRYPTOGRAPHY_ST_PW_CB *st = (CRYPTOGRAPHY_ST_PW_CB *)userdata;
+    st->called += 1;
+    st->maxsize = size;
+    if (st->length == 0) {
+        st->error = -1;
+        return 0;
```

Comment:
According to the docs "The callback must return the number of characters in the passphrase or 0 if an error occurred". Returning -1 appears to have been incorrect. Looking in the implementation of `PEM_read_bio_PrivateKey` (one of several functions that use this path) the cb's return value is used as a length and then checked to make sure it's not `<= 0`. So returning -1 is safe in practice (at least for that function), but not technically correct.